### PR TITLE
Rename some variables and make code more readable

### PR DESCRIPTION
**Pull request by nelson-liu**
This PR will modify/overwrite Min's code for the task of multiple choice question answering.

I figured I'd first clean up the file, so it's a bit more readable and I can more clearly understand what's to be done.

### CODE REVIEW

File: basic/model.py [hunk 1]
```diff
@@ -68,143 +96,180 @@ def __init__(self, config, scope, rep=True):
...
+                # [batch_size, max_num_sentences, max_sentence_size, 2*hidden_size]
+                (fw_outputs, bw_outputs), _ = bidirectional_dynamic_rnn(cell, cell, embedded_passage, passage_len, dtype='float', scope='encoded_passage')
+                # [batch_size, max_num_sentences, max_sentence_size, 2*hidden_size]
+                encoded_passage = tf.concat(3, [fw_outputs, bw_outputs])
+            self.tensor_dict['encoded_question'] = encoded_question
+            self.tensor_dict['encoded_passage'] = encoded_passage
 
         with tf.variable_scope("main"):
+            # The Attention Flow layer
+            # TODO: NOT SURE WHAT TO CALL P0 here / NOT SURE WHAT IT REPRESENTS.
```
Comment (matt-gardner) [inline basic/model.py]: Min's `p0` is what I call `final_merged_passage` here: https://github.com/allenai/deep_qa/blob/94177eb084fb70fa198e216a05f77e335d388ea0/deep_qa/models/reading_comprehension/bidirectional_attention.py#L152.

File: basic/model.py [hunk 2]
```diff
@@ -68,143 +96,180 @@ def __init__(self, config, scope, rep=True):
...
+            (fw_output_biencoder1, bw_output_biencoder1), _ = bidirectional_dynamic_rnn(first_cell, first_cell, modeled_passage_0, passage_len, dtype='float', scope='model_layer_biencoder_1')
+            # This encoding is used to get the start span index.
+            modeled_passage_1 = tf.concat(3, [fw_output_biencoder1, bw_output_biencoder1])
+
+            # The logits for the start span answer.
+            logits = get_logits([modeled_passage_1, p0], hidden_size, True, wd=config.wd, input_keep_prob=config.input_keep_prob,
+                                mask=self.passage_mask, is_train=self.is_train, func=config.answer_func, scope='logits1')
+
+            # TODO: NOT QUITE SURE WHAT'S GOING ON HERE EITHER
+            a1i = softsel(tf.reshape(modeled_passage_1, [batch_size, max_num_sentences * max_sentence_size, 2 * hidden_size]), tf.reshape(logits, [batch_size, max_num_sentences * max_sentence_size]))
```
Comment (matt-gardner) [inline basic/model.py]: Min's `a1i` is my `passage_weighted_by_predicted_span`, here: https://github.com/allenai/deep_qa/blob/94177eb084fb70fa198e216a05f77e335d388ea0/deep_qa/models/reading_comprehension/bidirectional_attention.py#L182-L185.

File: basic/model.py [hunk 3]
```diff
@@ -68,143 +96,180 @@ def __init__(self, config, scope, rep=True):
...
+            a1i = tf.tile(tf.expand_dims(tf.expand_dims(a1i, 1), 1), [1, max_num_sentences, max_sentence_size, 1])
+
+            # [batch_size, max_num_sentences, max_sentence_size, 2*hidden_size]
+            ((fw_end_encoder_output,
+              bw_end_encoder_output), _) = bidirectional_dynamic_rnn(cell_with_dropout, cell_with_dropout, tf.concat(3, [p0, modeled_passage_1, a1i, modeled_passage_1 * a1i]),
+                                                                     passage_len, dtype='float', scope='end_span_encoder')
+            end_span_modeled_passage = tf.concat(3, [fw_end_encoder_output, bw_end_encoder_output])
+            logits2 = get_logits([end_span_modeled_passage, p0], hidden_size, True, wd=config.wd, input_keep_prob=config.input_keep_prob,
```
Comment (matt-gardner) [inline basic/model.py]: `span_end_logits`?

File: basic/model.py [hunk 4]
```diff
@@ -68,143 +96,180 @@ def __init__(self, config, scope, rep=True):
...
+            modeled_passage_0 = tf.concat(3, [fw_output_biencoder0, bw_output_biencoder0])
+
+            # The 1st (second) bidirectional encoder in the modeling layer
+            # [batch_size, max_num_sentences, max_sentence_size, 2*hidden_size]
+            (fw_output_biencoder1, bw_output_biencoder1), _ = bidirectional_dynamic_rnn(first_cell, first_cell, modeled_passage_0, passage_len, dtype='float', scope='model_layer_biencoder_1')
+            # This encoding is used to get the start span index.
+            modeled_passage_1 = tf.concat(3, [fw_output_biencoder1, bw_output_biencoder1])
+
+            # The logits for the start span answer.
+            logits = get_logits([modeled_passage_1, p0], hidden_size, True, wd=config.wd, input_keep_prob=config.input_keep_prob,
```
Comment (matt-gardner) [inline basic/model.py]: `span_begin_logits`?

File: basic/model.py [hunk 5]
```diff
@@ -68,143 +96,180 @@ def __init__(self, config, scope, rep=True):
...
-        N, M, JX, JQ, VW, VC, d, W = \
-            config.batch_size, config.max_num_sents, config.max_sent_size, \
-            config.max_ques_size, config.word_vocab_size, config.char_vocab_size, config.hidden_size, \
-            config.max_word_size
-        JX = tf.shape(self.x)[2]
-        JQ = tf.shape(self.q)[1]
-        M = tf.shape(self.x)[1]
-        dc, dw, dco = config.char_emb_size, config.word_emb_size, config.char_out_size
+        batch_size = config.batch_size
+        max_num_sentences = config.max_num_sents
```
Comment (matt-gardner) [inline basic/model.py]: I'd probably remove the `max_` prefixes here; they don't really add anything.

**Review by matt-gardner — APPROVED** [review]
This is just renaming things, right?  Looks fine to me.

**Comment by nelson-liu** [comment]
yeah, it's just renaming things so I have a clearer idea of what's going on in the code. thanks for taking a look.

**Comment by nelson-liu** [comment]
i've made a separate branch from master called `mc_bidaf`, where i'll be merging this PR and future MC BiDAF-related PRs. I'll merge it tomorrow after I've successfully reproduced the results.

**Comment by nelson-liu** [comment]
oops, forgot to merge this. i just checked, and this version of bidaf gets more or less the same results a run with the original code. merging it now 
screen shot 2017-04-03 at 12 46 22 pm

**Pull request merged**